### DISCUSSION TITLE
"Upload Sketch" vs. "Upload the sketch using the selected programmer"

### DISCUSSION THREAD

**Discussion by RatGit** [Sloeber/arduino-eclipse-plugin#1643] [Q&A]
I have a project where I don't need the bootloader. Consequently, I have set the "build.ldscript" variable to "linker_scripts/gcc/flash_without_bootloader.ld". In the previous version of Sloeber simply clicking "Upload Sketch" would flash this image to the target device. Now it appears that I must use "Upload the sketch using the selected programmer" to do this? (Otherwise my code fails to run on the target.)

Have I correctly interpreted the new Sloeber behaviour?

**Comment by jantje** [comment]
>Have I correctly interpreted the new Sloeber behaviour?

I think so (This is assuming I correctly understood your question)
The Sloeber behavior of the upload button changed from "upload if no programmer is selected else program using the selected programmer" to "upload".
To program a board in the newer Sloeber use the added "Upload the sketch using the selected programmer" button.
This makes Sloeber behavior more compliant with Arduino IDE behavior.
